great_expectations/expectations/model_field_types.py

```diff
@@ -22,10 +22,12 @@
 ]
 
 ValueSetField = Annotated[
-    Union[Sequence, set, SuiteParameterDict],
-    pydantic.Field(
+    Union[List, SuiteParameterDict],
+    Field(
         title="Value Set",
         description=VALUE_SET_DESCRIPTION,
+        # Add type constraint to exclude strings and bytes
+        not_={"anyOf": [{"type": "string"}, {"type": "string", "format": "binary"}]},
```

Comment:
for reviewers - this prevents string and bytestring from coercing to a list. Should we skip this change and instead have an expectation validation error if someone passes a string instead of expected iterables (which is unlikely but possible)?
I am also not sure if we want to update schemas - can remove this addition and undo schema changes